trovi/api/serializers.py

```diff
@@ -255,6 +258,29 @@ def to_internal_value(self, data: dict[str, JSON]) -> dict[str, JSON]:
         return super(ArtifactVersionMetricsSerializer, self).to_internal_value(data)
 
 
+class ArtifactMetricsSerializer(serializers.Serializer):
+    """
+    Describes an artifact level metrics, which are related to events
+    """
+
+    class Meta:
+        read_only_fields = ["access_count", "unique_access_count"]
+
+    def to_representation(self, instance: ArtifactVersion) -> dict[str, JSON]:
+        unique_origins = set()
+        total_access_count = 0
+        for v in instance.versions.all():
+            origins = v.events.filter(event_type=ArtifactEvent.EventType.LAUNCH).values(
+                "event_origin"
+            )
+            unique_origins.update([o["event_origin"] for o in origins])
+            total_access_count += v.access_count
+        return {
+            "access_count": total_access_count,
+            "unique_access_count": len(unique_origins),
+        }
```

Comment:
`Artifact`s already have an `access_count` field which we can use here. I also checked, and I think it's possible to limit the calculation of `unique_access_count` to a single SQL query, similar to how you did it for `ArtifactVersion`

```suggestion
        return {
            "access_count": total_access_count,
            "unique_access_count": ArtifactEvent.objects.filter(
                artifact_version__artifact=instance,
                event_type=ArtifactEvent.EventType.LAUNCH,
            )
            .values("event_origin")
            .distinct()
            .count(),
        }
```